services/gateway/gateway-service.mjs

```diff
@@ -275,25 +294,39 @@ class GatewayService {
     return true;
   }
 
-  async broadcastResetUser(socket, uid, mapCoord, facing) {
-    socket.broadcast.emit('location', {uid, mapCoord, facing});
+  _getDistanceSquare(a,b){
```

Comment:
[Not required in this PR] Please consider moving the checks below into a library that can be used both in backend and frontend.